implement get all courses

teodorstrut opened the pull request:
I have added the basic functionality to get all the courses from the BE.
Tested with postman. **register**, **login**, and access `api/courses/` to get all courses

teodorstrut committed:
implement get all courses

I have added the basic functionality to get all the courses from the BE

teodorstrut committed:
updated route visibility

Changed the route to "/courses/all", and opened route for public access

teodorvoicencu commented on backend/src/main/java/ubb/courses/backend/controllers/course/CourseController.java:
```diff
@@ -1,18 +1,33 @@
+    @GetMapping("/all")
+    public ArrayList<CourseDTO> GetAllCourses(){
+        ArrayList<CourseDTO> courseDTOs = new ArrayList<>();
+        this.courseService.getAllCourses().forEach(course -> courseDTOs.add(courseConverter.createFrom(course)));
```
You can make those one line if you use map on the result array

teodorvoicencu commented on backend/src/main/java/ubb/courses/backend/controllers/course/ICourseController.java:
```diff
@@ -1,4 +1,9 @@
+import java.util.ArrayList;
+
 public interface ICourseController {
+    public ArrayList<CourseDTO> GetAllCourses();
```
Please use ResponseEntity. See the user controller to make an idea of it.

teodorvoicencu commented on backend/src/main/java/ubb/courses/backend/dtos/courses/CourseDTO.java:
```diff
@@ -0,0 +1,21 @@
+
+    private Date createdDate;
+
+    private Date modifiedDate;
```
Those properties are used only for Admin auditing, please remove them from the DTO

teodorvoicencu commented on backend/src/main/java/ubb/courses/backend/repositories/CourseRepository.java:
```diff
@@ -1,9 +1,13 @@
 package ubb.courses.backend.repositories;
 
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
```
This is not used, remove it

teodorstrut committed:
minor fixes

removed unused imports, updated return types in controller, made one-liner method

teodorvoicencu reviewed (approved):
LGTM
